### PR TITLE
[AFD WAF] Add exclusions to managed rule set configuration (2019-10-01)

### PR DESCRIPTION
**Pull request by eschwabe**
### Overview
* Add API Version 2019-10-01 for FrontDoor WAF API
* Add exclusions to managed rule set configuration at the rule set, rule group, and rule levels
* Updated examples with valid configurations

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [X] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [X] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [X] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/frontdoor/resource-manager/Microsoft.Network/stable/2019-10-01/webapplicationfirewall.json [hunk 1]
```diff
@@ -0,0 +1,876 @@
...
+        "selectorMatchOperator": {
+          "type": "string",
+          "enum": [
+            "Equals",
+            "Contains",
+            "StartsWith",
+            "EndsWith",
+            "EqualsAny"
+          ],
+          "description": "Comparison operator to apply to the selector when specifing which elements in the collection this exclusion applies to.",
```
Comment (nschonni) [inline specification/frontdoor/resource-manager/Microsoft.Network/stable/2019-10-01/webapplicationfirewall.json]:
```suggestion
          "description": "Comparison operator to apply to the selector when specifying which elements in the collection this exclusion applies to.",
```

File: specification/frontdoor/resource-manager/Microsoft.Network/stable/2019-10-01/examples/WafListManagedRuleSets.json [hunk 2]
```diff
@@ -0,0 +1,56 @@
+{
```
Comment (KrisBash) [inline specification/frontdoor/resource-manager/Microsoft.Network/stable/2019-10-01/examples/WafListManagedRuleSets.json]: For new API versions, please copy a prior reviewed API version as the first commit, and then commit the changes. It makes it easier to review deltas in Github natively. Thanks

File: specification/frontdoor/resource-manager/Microsoft.Network/stable/2019-10-01/network.json [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    "description": "The Microsoft Azure Network management API provides a RESTful set of web services that interact with Microsoft Azure Networks service to manage your network resources. The API has entities that capture the relationship between an end user and the Microsoft Azure Networks service.",
+    "version": "2019-03-01"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json",
+    "text/json"
```
Comment (KrisBash) [inline specification/frontdoor/resource-manager/Microsoft.Network/stable/2019-10-01/network.json]: Why is `text/json` necessary?

File: specification/frontdoor/resource-manager/Microsoft.Network/stable/2019-10-01/network.json [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Resource tags."
+        }
+      },
+      "description": "Tags object for patch operations."
+    }
+  },
```
Comment (KrisBash) [inline specification/frontdoor/resource-manager/Microsoft.Network/stable/2019-10-01/network.json]: A number of common type definitions were removed, when compared to 2019-07-01, such as ManagedIdentity, ProvisioningState, etc. Is that intentional?

File: specification/frontdoor/resource-manager/Microsoft.Network/stable/2019-10-01/webapplicationfirewall.json [hunk 5]
```diff
@@ -0,0 +1,812 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2019-03-01",
```
Comment (KrisBash) [inline specification/frontdoor/resource-manager/Microsoft.Network/stable/2019-10-01/webapplicationfirewall.json]: Version does not match folder name

File: specification/frontdoor/resource-manager/Microsoft.Network/stable/2019-10-01/webapplicationfirewall.json [hunk 6]
```diff
@@ -0,0 +1,812 @@
+{
```
Comment (KrisBash) [inline specification/frontdoor/resource-manager/Microsoft.Network/stable/2019-10-01/webapplicationfirewall.json]: This file seems identical to 2019-03-01/webapplicationfirewall.json. Are there other changes expected or is this just initializing the new version?

**Review by KrisBash — CHANGES_REQUESTED** [review]
Please clarify the intention, as the 2019-03-01 WAF swagger is being added unchanged to this folder.

File: specification/frontdoor/resource-manager/Microsoft.Network/stable/2019-10-01/examples/WafListManagedRuleSets.json (hunk 2 above)
Comment (eschwabe) [reply specification/frontdoor/resource-manager/Microsoft.Network/stable/2019-10-01/examples/WafListManagedRuleSets.json]: The first commit should be a copy of the 2019-03-01 API. The subsequent commits have the proposed changes for version 2019-10-01.

File: specification/frontdoor/resource-manager/Microsoft.Network/stable/2019-10-01/network.json (hunk 4 above)
Comment (eschwabe) [reply specification/frontdoor/resource-manager/Microsoft.Network/stable/2019-10-01/network.json]: The network.json was simply copied from the previous version. It has now been synchronized with the network APIs 2019-07-01 version.

File: specification/frontdoor/resource-manager/Microsoft.Network/stable/2019-10-01/webapplicationfirewall.json (hunk 6 above)
Comment (eschwabe) [reply specification/frontdoor/resource-manager/Microsoft.Network/stable/2019-10-01/webapplicationfirewall.json]: The first commit initializes the new version. The subsequent commits contain the proposed changes.

File: specification/frontdoor/resource-manager/Microsoft.Network/stable/2019-10-01/webapplicationfirewall.json (hunk 1 above)
Comment (eschwabe) [reply specification/frontdoor/resource-manager/Microsoft.Network/stable/2019-10-01/webapplicationfirewall.json]: Updated description.

File: specification/frontdoor/resource-manager/Microsoft.Network/stable/2019-10-01/network.json (hunk 3 above)
Comment (eschwabe) [reply specification/frontdoor/resource-manager/Microsoft.Network/stable/2019-10-01/network.json]: Synchronized network.json with 2019-07-01 from network APIs.

File: specification/frontdoor/resource-manager/Microsoft.Network/stable/2019-10-01/webapplicationfirewall.json (hunk 5 above)
Comment (eschwabe) [reply specification/frontdoor/resource-manager/Microsoft.Network/stable/2019-10-01/webapplicationfirewall.json]: See latest commits.

**Review by KrisBash — APPROVED** [review]
LGTM

**Review by tjprescott — APPROVED** [review]

**Pull request merged**